orchagent/vnetorch.cpp

```diff
@@ -2004,6 +2014,160 @@ bool VNetRouteOrch::delOperation(const Request& request)
     return true;
 }
 
+void VNetRouteOrch::attach(Observer* observer, const IpAddress& dstAddr)
+{
+    SWSS_LOG_ENTER();
+
+    auto insert_result = next_hop_observers_.emplace(dstAddr, VNetNextHopObserverEntry());
+    auto observerEntry = insert_result.first;
+
+    /* Create a new observer entry if no current observer is observing this
+     * IP address */
+    if (insert_result.second)
+    {
+        /* Find the prefixes that cover the destination IP */
+        for (auto route : syncd_routes_)
+        {
+            if (route.first.isAddressInSubnet(dstAddr))
+            {
+                SWSS_LOG_INFO("Prefix %s covers destination address",
+                    route.first.to_string().c_str());
+                observerEntry->second.routeTable.emplace(
+                    route.first,
+                    route.second
+                );
+            }
+        }
+    }
+
+    observerEntry->second.observers.push_back(observer);
+
+    SWSS_LOG_NOTICE("Attached next hop observer of route %s for destination IP %s",
+        observerEntry->second.routeTable.rbegin()->first.to_string().c_str(),
+        dstAddr.to_string().c_str());
+
+    for (const auto & route : observerEntry->second.routeTable)
+    {
+        VNetNextHopUpdate update = {
+            SET_COMMAND,
+            std::get<0>(route.second), // vnet name
+            dstAddr, // destination
+            route.first, // prefix
+            std::get<1>(route.second) // nexthop
+        };
+        observer->update(SUBJECT_TYPE_NEXTHOP_CHANGE, reinterpret_cast<void*>(&update));
```

Comment:
why not `static_cast`?

Reply:
Because `static_cast` more like some conversion between parent and child object, which will execute some extra instructions if it was needed. But `reinterpret_cast` only do memory copy, which had been used at Qt source code to implement its Publisher/Subscriber pattern.

Reply:
agree. since it's casting between pointers. my impression was that `static_cast` will always be safer.